Civi/Api4/Service/Spec/Provider/MessageTemplateGetSpecProvider.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+namespace Civi\Api4\Service\Spec\Provider;
+
+use Civi\Api4\Service\Spec\FieldSpec;
+use Civi\Api4\Service\Spec\RequestSpec;
+
+/**
+ * @service
+ * @internal
+ */
+class MessageTemplateGetSpecProvider extends \Civi\Core\Service\AutoService implements Generic\SpecProviderInterface {
+
+  /**
+   * @param \Civi\Api4\Service\Spec\RequestSpec $spec
+   */
+  public function modifySpec(RequestSpec $spec) {
+    $field = new FieldSpec('revert_id', 'MessageTemplate', 'Integer');
+    $field->setLabel(ts('Revert ID'))
+      ->setTitle(ts('Revert ID'))
+      ->setColumnName('id')
+      ->setDescription(ts('MessageID that this could revert to'))
+      ->setInputType('Select')
+      ->setReadonly(TRUE)
+      ->setFkEntity('MessageTemplate')
+      ->setSqlRenderer([__CLASS__, 'revertible']);
+    $spec->addFieldSpec($field);
+  }
+
+  /**
+   * @param string $entity
+   * @param string $action
+   *
+   * @return bool
+   */
+  public function applies($entity, $action) {
+    return $entity === 'MessageTemplate' && $action === 'get';
+  }
+
+  /**
+   * Callback for finding id of template to revert to
+   * Based on CRM_Admin_Page_MessageTemplates::__construct()
+   *
+   * @return string
+   */
+  public static function revertible(): string {
```

Comment:
does this have to be public static - I see the way it is being called is a bit funky so maybe the answer is yes - but protected non-static is better where possible

Reply:
I originally had it as not static but the tests failed objecting to it being called statically, and since the tests aren't calling it directly I assume that's how it gets called but didn't dig further.

Reply:
Yes it has to be `public static` as it's a callback for the APIv4 Query object outside this class.